plugins/woocommerce/src/Blocks/BlockTypes/AddToCartWithOptionsVariationSelector.php

```diff
@@ -17,16 +17,313 @@ class AddToCartWithOptionsVariationSelector extends AbstractBlock {
 	 */
 	protected $block_name = 'add-to-cart-with-options-variation-selector';
 
+	/**
+	 * Render variation label.
+	 *
+	 * @param string $attribute_name Name of the attribute.
+	 * @return string Rendered label HTML.
+	 */
+	protected function render_variation_label( $attribute_name ): string {
+		$label_id   = esc_attr( 'attribute_' . sanitize_title( $attribute_name ) );
+		$label_text = wc_attribute_label( $attribute_name );
+
+		$html = sprintf(
+			'<label class="wc-block-components-product-add-to-cart-attribute-label" for="%s">%s</label>',
+			$label_id,
+			esc_html( $label_text )
+		);
+		return $html;
+	}
+
+	/**
+	 * Render variation selector dropdown.
+	 *
+	 * @param WC_Product $product The product object.
+	 * @param string     $attribute_name Name of the attribute.
+	 * @param array      $options Available options for this attribute.
+	 * @return string Rendered dropdown HTML.
+	 */
+	protected function render_variation_selector( $product, $attribute_name, $options ): string {
+		$selected  = $this->get_selected_attribute_value( $product, $attribute_name );
+		$select_id = esc_attr( 'attribute_' . sanitize_title( $attribute_name ) );
+
+		$html = sprintf(
+			'<select id="%1$s"
+				class="wc-block-components-product-add-to-cart-attribute-select"
+				name="%1$s"
+				data-attribute_name="attribute_%2$s"
+				data-wc-on--change="actions.updateVariation">',
```

Comment:
Currently `actions.updateVariation` is not hooked to any action, no? Wondering if we should remove it for now or if there is a reason to keep it.

Reply:
Good idea! [cda54c1](https://github.com/woocommerce/woocommerce/pull/53947/commits/cda54c1e1f316cddaf7f4eb165462684f4208873)